tree/tree/src/TTree.cxx

```diff
@@ -809,7 +810,6 @@ TTree::TTree(const char* name, const char* title, Int_t splitlevel /* = 99 */)
    // FIXME: This is very annoying behaviour, we should
    //        be able to choose to not do this like we
    //        can with a histogram.
-   fDirectory = gDirectory;
```

Comment:
No it's doesn't. If you don't pass `nullptr` it will still add itself, so you need to fix up all call-sites to pass `nullptr`, while `TH1` has a global setting that disables adding-to-directories.